apps/extension/src/state/accounts.ts

```diff
@@ -47,12 +53,19 @@ export const selectedAccount = (state: AllSlices): Account | undefined => {
   const active = getActiveWallet(state);
   if (!active) return undefined;
 
-  const addr = getAddressByIndex(active.fullViewingKey, state.accounts.index);
+  const ephemeral = state.accounts.ephemeral;
+  const index = state.accounts.index;
+
+  const addr = ephemeral
+    ? getEphemeralByIndex(active.fullViewingKey, index)
+    : getAddressByIndex(active.fullViewingKey, index);
   const bech32Addr = bech32Address(addr);
 
   return {
     address: bech32Addr,
-    preview: getShortAddressByIndex(active.fullViewingKey, state.accounts.index),
-    index: state.accounts.index,
+    preview: ephemeral
+      ? bech32Addr.slice(0, 33) + '…'
```

Comment:
This isn't ideal. We should replace `get_short_address_by_index` with `get_short_address` so it can be used generically. Maybe @Valentine1898, you can assist with that?

And then on the next wasm crate deployment, we can update this line.